src/user/PreferencesController.ts

```diff
@@ -188,7 +190,25 @@ export class PreferencesController extends BaseController<BaseConfig, Preference
       };
       return ids;
     }, {});
-    this.update({ identities: { ...identities } });
+
+    let { selectedAddress } = this.state;
+    if (!Object.keys(identities).includes(selectedAddress)) {
+      selectedAddress = Object.keys(identities)[0];
+    }
+    this.update({ identities: { ...identities }, selectedAddress });
```

Comment:
The `updateIdentities` method has been updated to also reset the selected address, if the current selected address is no longer preset as an identity. This happens after every call to `updateIdentities`, so it may as well be built-in. This should be done in a separate PR.

Reply:
Done in #415